File: simuladores/simulador_memoria.py
import time


def main() -> None:
    bloques = []
    bloque_mb = 32
    repeticiones = 10

    for indice in range(repeticiones):
        bloques.append(bytearray(bloque_mb * 1024 * 1024))
        print(f"Memoria simulada: {(indice + 1) * bloque_mb} MB")
        time.sleep(1)

    time.sleep(5)


if __name__ == "__main__":
    main()

File: simuladores/simulador_memoria.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

/*
 * Simulador de consumo gradual de memoria para probar alertas de recursos.
 * Reserva bloques, los toca para forzar RSS y libera al finalizar.
 */
int main(void) {
    enum { BLOQUE_MB = 32, REPETICIONES = 10 };
    volatile unsigned char *bloques[REPETICIONES] = {0};
    for (int i = 0; i < REPETICIONES; ++i) {
        size_t bytes = (size_t)BLOQUE_MB * 1024 * 1024;
        bloques[i] = malloc(bytes);
        if (bloques[i] == NULL) break;
        for (size_t pagina = 0; pagina < bytes; pagina += 4096) {
            bloques[i][pagina] = (unsigned char)i;
        }
        printf("Memoria simulada: %d MB\n", (i + 1) * BLOQUE_MB);
        sleep(1);
    }
    sleep(5);
    for (int i = 0; i < REPETICIONES; ++i) free((void *)bloques[i]);
    return 0;
}
